src/robot/utils/robottypes.py

```diff
@@ -33,6 +50,50 @@ def is_str_like(item):
     return isinstance(item, (basestring, UserString, String))
 
 
+if PY3:
+    _integer = int
+    _number = int, float
+    _bytes = bytes
+    _bytes_like = bytes, bytearray
+    _string = str
+    _string_like = str, UserString
+else:
+    _integer = int, long
+    _number = int, long, float
+    _bytes = str
+    _bytes_like = str, bytearray, UserString, MutableString, String
+    _string = basestring
+    _string_like = basestring, UserString, MutableString, String
+
+
+def is_integer(item):
+    return isinstance(item, _integer)
+
+def is_number(item):
+    return isinstance(item, _number)
+
+def is_bytes(item):
+    return isinstance(item, _bytes)
+
+def is_bytes_like(item):
+    return isinstance(item, _bytes_like)
+
+def is_string(item):
+    return isinstance(item, _string)
+
+def is_string_like(item):
+    return isinstance(item, _string_like)
+
+if PY3:
+    is_unicode = is_string
+    is_unicode_like = is_string_like
+else:
+    def is_unicode(item):
+        return isinstance(item, unicode)
+
+    is_unicode_like = is_unicode
+
+
```

Comment:
This if/else logic looks overly complicated. What about adding separate `python2types` and `python3types` modules and having this in `robottypes`:

```python
if PY3:
    from python3types import is_string, is_integer, ...
else:
    from python2types import is_string, is_integer, ...
```

This approach has worked great when dealing with differences between CPython, Jython and IronPython.